Jenkinsfile.upstream-operator-sync

```diff
@@ -0,0 +1,168 @@
+        jdk 'kie-jdk11'
+    }
+    options {
+        buildDiscarder logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '10')
+        timeout(time: 90, unit: 'MINUTES')
+    }
+    environment {
+        KOGITO_CI_EMAIL_TO = credentials("${JENKINS_EMAIL_CREDS_ID}")
+        OPENSHIFT_INTERNAL_REGISTRY = 'image-registry.openshift-image-registry.svc:5000'
+
+        // Use buildah container engine in this pipeline
+        CONTAINER_ENGINE="podman"
+        CONTAINER_TLS_OPTIONS="--tls-verify=false"
+    }
+    stages {
+        stage('Initialize') {
+            steps {
+                script{
+                    checkout(githubscm.resolveRepository("${env.REPO_NAME}", "${env.GIT_AUTHOR}", "${BUILD_BRANCH_NAME}", false))
+                }
+            }
+        }
+        stage('Check if Kogito operator has changed') {
+            steps {
+                sh 'make update-dependencies'
+                script {
+                    goModModified = githubscm.isThereAnyChanges()
+                }
+            }
+        }
+        stage('Test Kogito Operator') {
+            when {
+                expression {
+                    return goModModified
+                }
+            }
+            steps {
+                sh 'make test'
+            }
+        }
+        stage('Build Kogito Operator') {
+            when {
+                expression {
+                    return goModModified
+                }
+            }
+            steps {
+                sh "make BUILDER=${CONTAINER_ENGINE}"
+            }
+        }
+        stage('Push Operator Image to Openshift Registry') {
+            when {
+                expression {
+                    return goModModified
+                }
+            }
+            steps {
+                withCredentials([string(credentialsId: "${env.OPENSHIFT_API_KEY}" , variable: 'OPENSHIFT_API')]) {
+                    withCredentials([usernamePassword(credentialsId: "${env.OPENSHIFT_CREDS_KEY}", usernameVariable: 'OC_USER', passwordVariable: 'OC_PWD')]) {
+                        sh "oc login --username=${OC_USER} --password=${OC_PWD} --server=${OPENSHIFT_API} --insecure-skip-tls-verify"
+                    }
+                }
+                withCredentials([string(credentialsId: "${env.OPENSHIFT_REGISTRY_KEY}" , variable: 'OPENSHIFT_REGISTRY')]) {
+                    sh """
+                        set +x && ${CONTAINER_ENGINE} login -u jenkins -p \$(oc whoami -t) ${CONTAINER_TLS_OPTIONS} ${OPENSHIFT_REGISTRY}
+                        source ./hack/env.sh && TAG_OPERATOR=\$(getOperatorVersion)
+                        ${CONTAINER_ENGINE} tag quay.io/kiegroup/${env.OPERATOR_IMAGE_NAME}:\${TAG_OPERATOR} ${OPENSHIFT_REGISTRY}/openshift/${env.OPERATOR_IMAGE_NAME}:${operatorTagVersion}
+                        ${CONTAINER_ENGINE} push ${CONTAINER_TLS_OPTIONS} ${OPENSHIFT_REGISTRY}/openshift/${env.OPERATOR_IMAGE_NAME}:${operatorTagVersion}
```

Comment:
I think the `operatorTagVersion` should include the branch name to avoid concurrent run of the pipeline and override of the operator in OCP, as here we did not lock as we do for the [BDD tests](https://github.com/kiegroup/rhpam-kogito-operator/pull/22/files#diff-41006ea6088c261a59626783caa713069d3040016a59815ad9ede11587a9a108R93)

Which you could make it dynamic with a method, keeping the original `operatorTagVersion` and then calling
```
String getOperatorVersion() {
    return "${operatorTagVersion}-${env.BUILD_BRANCH_NAME}"
}
```